src/auto_pickup.cpp

```diff
@@ -343,9 +345,9 @@ void auto_pickup::show( const std::string &custom_name, bool is_autopickup )
             if( g->u.name != "" ) {
                 save_character();
             }
-        } else {
-            create_rules();
         }
+
+        ready = false;
```

Comment:
Looks very vague. Ready for what? Probably `invalidate_map_items()` could help (a pair for `refresh_map_items()`).